### PR TITLE
Replace slowDown with setSpeed in base env [Merge only if this makes sense]

### PR DESCRIPTION
**Pull request by fywu85**
Replacing `slowDown` with `setSpeed` in base env is required for `speed_mode=0` to be truly _all checks off_. 

**TL;DR** Merge this ONLY IF we think there should not be any fail-safe mechanism under speed mode zero. My personal preference is to merge this, as this is consistent with the literal definition of zero speed mode. Plus, users who want fail-safe features would not set speed mode to zero anyway.

This is because that `slowDown` will issue an emergency braking of -9 m/s2 when the vehicle is about to collide (even if the speed mode is set to zero), while `setSpeed` will _faithfully and immediately_ execute the prescribed speed under speed mode zero (even if the command will result in a collision). 

Besides, under all other speed modes, the two commands seem to be interchangeable. `slowDown` with a 1 ms execution time and `setSpeed` seem to be behave exactly the same if there is no emergent collision. 

As a last corner case, they two commands are almost the same in speed mode zero, with `slowDown` setting a speed _slightly_ smaller (like 0.01 m/s) than the target speed.

### CODE REVIEW

**Comment by eugenevinitsky** [comment]
Okay, so the question is then, if speed mode is not set to zero, will emergency de-accelerations still be applied? We should write a test for this.

**Comment by fywu85** [comment]
No, `setSpeed` does not support emergency deccel. We could get this around by conditioning `apply_acceleration` routine on the speed mode. So the question becomes: do we want to do that?

**Comment by AboudyKreidieh** [comment]
As long as 0 speed mode is the only speed mode we use that'll stop respecting the bounds on accel/decel then I'm okay with this. This might be something to unit test.

**Comment by AboudyKreidieh** [comment]
as is it looks good to me. @eugenevinitsky if you have no objections, I'll approve

**Comment by AboudyKreidieh** [comment]
Doesn’t look like we’re merging, so closing
